awscli/examples/ecr-public/set-repository-policy.rst

```diff
@@ -0,0 +1,118 @@
+                ]
+            }
+        ]
+    }
+
+Output::
+
+    {
+        "registryId": "12345678901",
+        "repositoryName": "project-a/nginx-web-app",
+        "policyText": "{\n  \"Version\" : \"2008-10-17\",\n  \"Statement\" : [ {\n    \"Sid\" : \"allow public pull\",\n    \"Effect\" : \"Allow\",\n    \"Principal\" : \"*\",\n    \"Action\" : [ \"ecr:BatchCheckLayerAvailability\", \"ecr:BatchGetImage\", \"ecr:GetDownloadUrlForLayer\" ]\n  } ]\n}"
+    }
+
+For more information, see `Setting a repository policy statement <https://docs.aws.amazon.com/AmazonECR/latest/public/public-repository-policy-examples.html>`__ in the *Amazon ECR Public User Guide*.
+
+**Example 2: To set repository policy to allow IAM user within your account to push images**
+
+The following ``set-repository-policy`` allow an IAM user within your account to push images using the cli input file://my-repository-policy.json in your AWS account. ::
+
+    aws ecr-public set-repository-policy \
+        --repository-name project-a/nginx-web-app \
+        --policy-text file://my-repository-policy.json 
+
+Contents of ``my-repository-policy.json``::
+
+    {
+        "Version": "2008-10-17",
+        "Statement": [
+            {
+                "Sid": "AllowPush",
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": [
+                        "arn:aws:iam::account-id:user/push-pull-user-1",
+                        "arn:aws:iam::account-id:user/push-pull-user-2"
+                    ]
+                },
+                "Action": [
+                    "ecr-public:BatchCheckLayerAvailability",
+                    "ecr-public:PutImage",
+                    "ecr-public:InitiateLayerUpload",
+                    "ecr-public:UploadLayerPart",
+                    "ecr-public:CompleteLayerUpload"
+                ]
+            }
+        ]
+    }
+
+Output::
+
+    {
+        "registryId": "12345678901",
+        "repositoryName": "project-a/nginx-web-app",
+        "policyText": "{\n  \"Version\" : \"2008-10-17\",\n  \"Statement\" : [ {\n    \"Sid\" : \"AllowPush\",\n    \"Effect\" : \"Allow\",\n    \"Principal\" : {\n      \"AWS\" : [ \"arn:aws:iam::12345678901:user/admin\", \"arn:aws:iam::12345678901:user/eksuser1\" ]\n    },\n    \"Action\" : [ \"ecr-public:BatchCheckLayerAvailability\", \"ecr-public:PutImage\", \"ecr-public:InitiateLayerUpload\", \"ecr-public:UploadLayerPart\", \"ecr-public:CompleteLayerUpload\" ]\n  } ]\n}"
+    }
+
+For more information, see `Setting a repository policy statement <https://docs.aws.amazon.com/AmazonECR/latest/public/public-repository-policy-examples.html>`__ in the *Amazon ECR Public User Guide*.
+
+**Example 3: To set repository policy to allow IAM user from different account to push images**
+
+The following ``set-repository-policy`` example allow a specific account to push images using cli input file://my-repository-policy.json in your AWS account. ::
```

Comment:
```suggestion
The following ``set-repository-policy`` example allows a specific account to push images using cli input file://my-repository-policy.json in your AWS account. ::
```

The connection between pushing images and the CLI input file is unclear. The repository policy doesn't add persmissions to use CLI input files; the task is accomplished using a CLI input file.